packages/botonic-plugin-nlu/src/nlu.js

```diff
@@ -0,0 +1,154 @@
+      mode: 'node',
+      uri: `${path.join(process.cwd(), NLU_PATH, MODELS_DIRNAME)}/`
+    }
+  }
+}
+
+function loadOption(lang, env) {
+  let nlu = {}
+  try {
+    if (env.mode === 'prod' || env.mode === 'dev') {
+      nlu.nluData = axios({
+        url: `${env.uri}${lang}/${NLU_DATA_FILENAME}`
+      })
+      nlu.model = tf.loadLayersModel(`${env.uri}${lang}/${MODEL_FILENAME}`)
+    } else {
+      nlu.nluData = JSON.parse(
+        readFile(`${env.uri}/${lang}/${NLU_DATA_FILENAME}`)
+      )
+      nlu.model = tf.loadLayersModel(
+        `file://${env.uri}${lang}/${MODEL_FILENAME}`
+      )
+    }
+  } catch (e) {
+    console.log('Cannot retrieve NLU Information', e)
+  }
+  return nlu
+}
+
+export class NLU {
+  constructor(options) {
+    return (async () => {
+      this.env = await resolveEnv()
+      this.langs = []
+      this.nlus = {}
+      for (let config of options) {
+        this.langs.push(config.LANG)
+        let { nluData, model } = loadOption(config.LANG, this.env)
+        this.nlus[config.LANG] = {
+          lang: config.LANG,
+          model: model,
+          nluData: nluData
+        }
+      }
+      await Promise.all([
+        ...Object.values(this.nlus).map(nlu => nlu.model),
+        ...Object.values(this.nlus).map(nlu => nlu.nluData)
+      ])
+      for (let [k, v] of Object.entries(this.nlus)) {
+        let nluData = await v.nluData
+        let { intentsDict, maxSeqLength, vocabulary } =
+          this.env.mode === 'node' ? nluData : nluData.data
+        let model = await v.model
+        let { lang } = v
+        this.nlus[k] = {
+          lang: lang,
+          model: model,
+          intents: intentsDict,
+          maxSeqLength: maxSeqLength,
+          tokenizer: new Tokenizer(vocabulary)
+        }
+      }
+      return this
+    })()
+  }
+
+  predict(userInput, nlu) {
+    let sequences = nlu.tokenizer.samplesToSequences(userInput)
+    let paddedSequences = []
+    let predictions = []
+    for (let sequence of sequences) {
+      paddedSequences.push(
+        padSequences([sequence], nlu.maxSeqLength).dataSync()
+      )
+    }
+    for (let paddedSequence of paddedSequences) {
+      predictions.push(
+        nlu.model.predict(tf.tensor([paddedSequence])).dataSync()
+      )
+    }
+    return predictions
+  }
+
+  getIntent(prediction, nlu) {
+    let results = {}
+    results.intents = []
+    prediction.map((prob, i) => {
+      let res = {}
+      res['intent'] = `${nlu.intents[i]}`
+      res['prob'] = prob
+      results.intents.push(res)
+    })
+    let index = prediction.indexOf(Math.max(...prediction))
+    results.intent = nlu.intents[index]
+    results.confidence = prediction[index]
+    results.language = nlu.lang
+    return results
+  }
+
+  getIntents(userInput) {
+    let lang = detectLang(userInput, this.langs)
+    let nlu = this.nlus[lang]
+    let predictions = this.predict(userInput, nlu)
+    let results = []
+    for (let prediction of predictions) {
```

Comment:
try using return predictions.map(p-> getIntent).map(result-> resultintents.sort)

Reply:
all loops with comments refactored in the new code